File: python/371_sum_of_two_integers.py
class Solution:
    def getSum(self, a: int, b: int) -> int:
        mask = 0xffffffff
        s = (a ^ b) & mask
        c = a & b
        while c!=0:
            a = s
            b = (c << 1) & mask
            s = (a ^ b) & mask
            c = a & b
        if s & 0x80000000:
            s -= 0x100000000
        return s

sol = Solution()
a = int(input())
b = int(input())
print(sol.getSum(a, b))

File: golang/371_sum_of_two_integers.go
package main

import "fmt"

func getSum(a int, b int) int {
	if a == 0 {
		return b
	}
	if b == 0 {
		return a
	}

	for b != 0 {
		c := a & b
		a = a ^ b
		b = c << 1
	}

	return a
}

func main() {
	var a, b int
	fmt.Scanf("%d %d", &a, &b)
	// fmt.Println("value::: ", a, b)
	fmt.Println(getSum(a, b))
}
